### ISSUE TITLE
VariablesToExport and PowerShell Performance

### ISSUE THREAD

**Issue by tommymaynard** [MicrosoftDocs/windowsserverdocs#2730]
This document makes no mention of VariablesToExport. Should we assume that this should also use an empty array `@()` and not a wildcard character `*` in order to not negatively impact PowerShell performance? I did want to note that I am aware of the comments inside a module manifest file (psd1). Those do indicate to use an empty array for FunctionsToExport, CmdletsToExport, and AliasesToExport while VariablesToExport does not. While this may answer the question, it might be nice to see a bit more about it in the online documents. That (1) it's not necessary to do this for VariablesToExport, and (2), the reason as to why.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e04eb8e0-6d57-6d4f-6d89-9fc91cd56883
* Version Independent ID: 9b127bf9-6a9e-957d-2c92-823fc82baa26
* Content: [PowerShell module authoring considerations](https://docs.microsoft.com/en-us/windows-server/administration/performance-tuning/powershell/module-authoring-considerations)
* Content Source: [WindowsServerDocs/administration/performance-tuning/powershell/module-authoring-considerations.md](https://github.com/MicrosoftDocs/windowsserverdocs/blob/master/WindowsServerDocs/administration/performance-tuning/powershell/module-authoring-considerations.md)
* Product: **windows-server-threshold**
* Technology: **performance-tuning-guide**
* GitHub Login: @lzybkr
* Microsoft Alias: **JasonSh**

**Comment by lzybkr** [comment]
tl;dr: `VariablesToExport` has almost no impact on performance regardless of what you specify or don't.

Command discovery has a big impact on PowerShell performance which is why there is a cache. Building the cache relies on Export entries other than `VariablesToExport`.

Because `VariablesToExport` is ignored during command discovery, it's value does not impact the performance of command discover.

That said, it certainly isn't bad for performance to specify an empty array or avoid the use of a wildcard.

**Comment by tommymaynard** [comment]
This reply was great -- thank you! I still think the help document **might** benefit from making this clarification, as to avoid answering the same question again. Appreciate your time!

**Comment by tommymaynard** [comment]
Closing issue...
